### DISCUSSION TITLE
Your own pass for GBuffer for v7

### DISCUSSION THREAD

**Discussion by AndreyPrakhov** [pmndrs/postprocessing#657] [Q&A]
Hi. Please tell me how to add my own pass to GBuffer for v7? Let's say some GBuffer.COLOR2 where the base color will be rendered into the texture

**Comment by vanruesc** [comment]
Hi,

the `GeometryPass` uses a [GBufferConfig](https://github.com/pmndrs/postprocessing/blob/32819bd156a323c2005241ca0692d754c38745b4/src/utils/GBufferConfig.ts) to manage its internal render target. This config can be modified via https://github.com/pmndrs/postprocessing/blob/32819bd156a323c2005241ca0692d754c38745b4/src/passes/GeometryPass.ts#L114 or by providing a custom `gBufferConfig` through the constructor options: https://github.com/pmndrs/postprocessing/blob/32819bd156a323c2005241ca0692d754c38745b4/src/passes/GeometryPass.ts#L185

Additionally, you'll have to modify three's built-in materials with custom shader code to render to the additional gBuffer textures. You can check [ShaderChunkExtensions](https://github.com/pmndrs/postprocessing/blob/32819bd156a323c2005241ca0692d754c38745b4/src/shader-chunks/ShaderChunkExtensions.ts) for reference.

Please not that this feature hasn't been tested extensively, so there might be bugs.
